core/tests/ledger_cleanup.rs

```diff
@@ -502,14 +500,10 @@ mod tests {
                                 break;
                             }
                         } else {
-                            let mut slot_id = start_slot;
+                            let slot_id = start_slot;
                             if slot_id > 0 {
-                                for shred in shreds_with_parent.iter_mut() {
-                                    shred.set_slot(slot_id);
-                                    if shred.index() as u64 == shreds_per_slot - 1 {
-                                        slot_id += 1;
-                                    }
-                                }
+                                let (shreds_with_parent, _) = make_many_slot_shreds(
+                                    slot_id, batch_size_slots, shreds_per_slot);
```

Comment:
So by removing the update-in-place by re-generating shreds, this change will, unfortunately, ruin the original purpose of the code :(.

Reply:
Offline discussed with @behzadnouri.  Our conclusion is that we want to move forward with the merkle shreds PRs and separate the benchmark part in the test to an actual benchmark as it is not possible to deprecate the set_slot API while making the test able to update in place.  In addition, the merkle shred PR does not break the "test" while it makes the test be used as a shred insertion benchmark.